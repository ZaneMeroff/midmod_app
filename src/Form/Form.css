input {
  height: 30px;
  margin: 1%;
}

button {
  height: 40px;
  width: 60px;
}
