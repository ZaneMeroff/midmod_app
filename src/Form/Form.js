import React, { Component } from 'react';
import './Form.css'

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  updateChange = event => {
    this.setState({[event.target.name]: event.target.value})
  }

  submitRes = () => {
    this.props.postResOnDOM({
      id: 4,
      name: this.state.name,
      date: this.state.date,
      time: this.state.time,
      number: this.state.number
      })
    this.setState({id: '', name: '', date: '', time: '', number: ''})
  }

  validateInputs = () => {
    if (!this.state.name || !this.state.date || !this.state.time || !this.state.number) {
      window.alert('Please enter all requested information')
    } else {
      this.props.postToAPI(this.state.name, this.state.date, this.state.time, this.state.number)
      this.clearInputs()
    }
  }

  clearInputs = () => {
    this.setState({name: '', date: '', time: '', number: ''})
  }

  render() {
    return (
      <section>
        <input
          name='name'
          value={this.state.name}
          onChange={this.updateChange}
          placeholder='name...'
          type='text'
        />
        <input
          name='date'
          value={this.state.date}
          onChange={this.updateChange}
          placeholder='date...'
          type='text'
        />
        <input
          name='time'
          value={this.state.time}
          onChange={this.updateChange}
          placeholder='time...'
          type='text'
        />
        <input
          name='number'
          value={this.state.number}
          onChange={this.updateChange}
          placeholder='number...'
          type='number'
        />
        <button onClick={this.validateInputs}>Submit Res!</button>
      </section>
    )
  }
}

export default Form;
