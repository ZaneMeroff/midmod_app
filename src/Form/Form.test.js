import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import { shallow } from 'enzyme';

describe('Form', () => {
  let wrapper;
  let mockPostToAPI = jest.fn();
  let mockRes = [{
    name: 'fakeName',
    date: 'fakeDate',
    time: 'fakeTime',
    number: 'fakeNumber',
    id: 'fakeId',
  }]

  beforeEach(() => {
    wrapper = shallow(
      <Form
        postToAPI={mockPostToAPI}
        res={mockRes}
       />
     )
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Form />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when clearInputs is invoked', () => {
    wrapper.instance().clearInputs()
    expect(wrapper.state()).toEqual({name: '', date: '', time: '', number: ''})
  })

  it('should update state when updateChange is invoked', () => {
    let mockEvent = {
      target: {
        name: 'mockName',
        date: 'mockDate',
        time: 'mockTime',
        number: 'mockNumber',
        id: 'mockId',
      }
    }
    wrapper.instance().updateChange(mockEvent)
    expect(wrapper.state('name')).toEqual('mockName')
  })

})
