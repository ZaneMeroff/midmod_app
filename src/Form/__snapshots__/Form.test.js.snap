// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<section>
  <input
    name="name"
    onChange={[Function]}
    placeholder="name..."
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="date..."
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="time..."
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="number..."
    type="number"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Submit Res!
  </button>
</section>
`;
